src/components/inputs/Search/SearchBar/SearchBar.tsx

```diff
@@ -1,20 +1,51 @@
 import { ReactComponent as SearchIcon } from '../../../icons/Search.svg';
-import { ReactComponent as ArrowUpRight } from '../../../icons/ArrowUpRight.svg';
 import './styles.css';
 import { SearchBarProps } from '../../../../types';
+import React, { useState } from 'react';
 
 function SearchBar(props: SearchBarProps) {
-  const { buttonAction, placeholderText, headerText, footerText } = props;
+  const {
+    buttonAction,
+    placeholderText,
+    headerElement,
+    footerText,
+    availability,
+  } = props;
+
+  const [searchBarText, setSearchBarText] = useState('');
+
+  function onHandleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    setSearchBarText(e.target.value);
+    if (e.target.value == '') {
+      buttonAction(e.target.value);
+    }
```

Comment:
why would we only call the action when the value is empty? we also probably want to trim the value of any whitespace with `trim()`

Reply:
availabilityCheck func has a return for when its an empty string. Sets the placeholder and availability state which changes how the header and border of the searchbar renders

Reply:
i don't think we should even allow that state. if the text is empty, default state is shown

Reply:
not when you erase a name. Consider the following: search > name is taken > user deletes name to search again > searchbar still shows {previous name} is taken